Source/WebCore/page/ResizeObservation.cpp

```diff
@@ -118,6 +119,19 @@ std::optional<ResizeObservation::BoxSizes> ResizeObservation::elementSizeChanged
             return currentSizes;
         break;
     case ResizeObserverBoxOptions::ContentBox:
+        if (isObserverForContainIntrinsicSize) {
+            // To make sure disconnected target get deliveried, so it could disconnectFromResizeObservers and lastRemembedSize will be clear.
+            if (!m_target->isConnected())
+                return currentSizes;
+            if (!m_target->renderBox())
+                return { };
+            auto lastRememberedSize = m_target->lastRememberedSize();
+            if (!lastRememberedSize)
+                return currentSizes;
+            if (LayoutSize(lastRememberedSize->inlineSize(), lastRememberedSize->blockSize()) != currentSizes->contentBoxLogicalSize)
+                return currentSizes;
+            return { };
+        }
```

Comment:
Yes, this is necessary. There are some differences between them. For instance,
1. `lastRememberedSize` is not always same to `m_lastObservationSizes`, we skip updating `lastRememberedSize` when it is size containment.
2. RO would not trigger delivery if the size is same to the initial size. But c-i-s should, for it need to create `lastRememberedSize`.
3. When target is removed from DOM tree, c-i-s needs to clear  `lastRememberedSize` and disconnect, but RO not.

Reply:
Maybe I should find a way to specify the size comparing function?

Reply:
That's how I did these in Firefox:

1. Just observe or unobserve the element when the condition for updating the last remembered size changes
2. If you mean that no observation is triggered when observing a 0x0 element, the CSSWG resolved to change that in general, so c-i-s shouldn't need a special behavior
3. I handled this in `Element::UnbindFromTree` (probably WebKit has something similar): unobserving and observing the element schedules a notification, even if the size was already 0x0. Then the callback checks if the element is disconnected, if so it removes the last remembered size.

Reply:
Thanks, @Loirooriol ! Let me have a try of this!
Looks like we need to address the initial size issue in RO first.
Just uploaded the patch https://github.com/WebKit/WebKit/pull/8839

Reply:
Sorry for not updating recently, but I'm stuck in this issue https://github.com/WebKit/WebKit/pull/9112#issuecomment-1403885997

Reply:
OK, 2 is now fixed:)
As to 1 and 3, instead of unobserving the target, we update the m_lastObservationSizes in the new patch.
For 1, set m_lastObservationSizes to lastRememberedSize.
For 3, set update m_lastObservationSizes to -1x-1, to ensure that the disconnected target could be delivered.